lib/service/connect.go

```diff
@@ -76,6 +85,22 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 	}
 }
 
+func (process *TeleportProcess) compareVersions(resp *proto.PingResponse) error {
+	serverVersion, err := semver.NewVersion(resp.ServerVersion)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	teleportVersion, err := semver.NewVersion(teleport.Version)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if math.Abs(float64(serverVersion.Major-teleportVersion.Major)) > 1 {
+		process.log.Warnf("Only a two major version difference between auth client and auth server are supported. Auth server is v%s",
```

Comment:
The message is still confusing I think.

What about `process.log.Warnf("Only versions %s and %s are supported, auth server is %s.", teleportVersion.Major, teleportVersion.Major-1, serverVersion.Major)`